pymatgen/io/lobster/inputs.py

```diff
@@ -301,27 +271,53 @@ def from_dict(cls, dct: dict) -> Self:
         Returns:
             Lobsterin
         """
-        return cls({k: v for k, v in dct.items() if k not in ["@module", "@class"]})
+        return cls({key: val for key, val in dct.items() if key not in {"@module", "@class"}})
+
+    def _get_nbands(self, structure: Structure) -> int:
+        """Get number of bands."""
+        if self.get("basisfunctions") is None:
+            raise ValueError("No basis functions are provided. The program cannot calculate nbands.")
+
+        basis_functions: list[str] = []
+        for string_basis in self["basisfunctions"]:
+            string_basis_raw = string_basis.strip().split(" ")
+            while "" in string_basis_raw:
+                string_basis_raw.remove("")
+            for _idx in range(int(structure.composition.element_composition[string_basis_raw[0]])):
+                basis_functions.extend(string_basis_raw[1:])
+
+        num_basis_functions = 0
+        for basis in basis_functions:
+            if "s" in basis:
+                num_basis_functions += 1
+            elif "p" in basis:
+                num_basis_functions += 3
+            elif "d" in basis:
+                num_basis_functions += 5
+            elif "f" in basis:
+                num_basis_functions += 7
+
+        return int(num_basis_functions)
 
     def write_INCAR(
         self,
-        incar_input: str = "INCAR",
-        incar_output: str = "INCAR.lobster",
-        poscar_input: str = "POSCAR",
-        isym: int = -1,
+        incar_input: PathLike = "INCAR",
+        incar_output: PathLike = "INCAR.lobster",
+        poscar_input: PathLike = "POSCAR",
+        isym: Literal[-1, 0] = -1,
         further_settings: dict | None = None,
-    ):
-        """Will only make the run static, insert nbands, make ISYM=-1, set LWAVE=True and write a new INCAR.
-        You have to check for the rest.
+    ) -> None:
+        """Write INCAR file. Will only make the run static,insert NBANDS,
+        set ISYM=-1, LWAVE=True and you have to check for the rest.
 
         Args:
-            incar_input (str): path to input INCAR
-            incar_output (str): path to output INCAR
-            poscar_input (str): path to input POSCAR
-            isym (int): isym equal to -1 or 0 are possible. Current Lobster version only allow -1.
+            incar_input (PathLike): path to input INCAR
+            incar_output (PathLike): path to output INCAR
+            poscar_input (PathLike): path to input POSCAR
+            isym (int): ISYM equals -1 or 0 are possible. Current LOBSTER version only allow -1.
```

Comment:
```suggestion
            isym (int): ISYM equals -1 or 0 are possible.
```

The last part can be removed now.

Reply:
OK!